lib/ibandit/german_details_converter.rb

```diff
@@ -0,0 +1,1270 @@
+        '50080000_0000101010' => '0090003500',
+        '50040000_0000101010' => '0311011100',
+        '37040044_0000102030' => '0222344400',
+        '86080000_0000121200' => '0480375900',
+        '66280053_0000121212' => '0625242400',
+        '16080000_0000123456' => '0012345600',
+        '29080010_0000124124' => '0107502000',
+        '37080040_0000182002' => '0216603302',
+        '12080000_0000212121' => '4050462200',
+        '37080040_0000300000' => '0983307900',
+        '37040044_0000300000' => '0300000700',
+        '37080040_0000333333' => '0270330000',
+        '38040007_0000336666' => '0105232300',
+        '55040022_0000343434' => '0217900000',
+        '85080000_0000400000' => '0459488501',
+        '37080040_0000414141' => '0041414100',
+        '38040007_0000414141' => '0108000100',
+        '20080000_0000505050' => '0500100600',
+        '37080040_0000555666' => '0055566600',
+        '20080000_0000666666' => '0900732500',
+        '30080000_0000700000' => '0800005000',
+        '70080000_0000700000' => '0750055500',
+        '70080000_0000900000' => '0319966601',
+        '37080040_0000909090' => '0269100000',
+        '38040007_0000909090' => '0119160000',
+        '70080000_0000949494' => '0575757500',
+        '70080000_0001111111' => '0448060000',
+        '70040041_0001111111' => '0152140000',
+        '10080000_0001234567' => '0920192001',
+        '38040007_0001555555' => '0258266600',
+        '76040061_0002500000' => '0482146800',
+        '16080000_0003030400' => '4205227110',
+        '37080040_0005555500' => '0263602501',
+        '75040062_0006008833' => '0600883300',
+        '12080000_0007654321' => '0144000700',
+        '70080000_0007777777' => '0443540000',
+        '70040041_0007777777' => '0213600000',
+        '64140036_0008907339' => '0890733900',
+        '70080000_0009000000' => '0319966601',
+        '61080006_0009999999' => '0202427500',
+        '12080000_0012121212' => '4101725100',
+        '29080010_0012412400' => '0107502000',
+        '34280032_0014111935' => '0645753800',
+        '38040007_0043434343' => '0118163500',
+        '30080000_0070000000' => '0800005000',
+        '70080000_0070000000' => '0750055500',
+        '44040037_0111111111' => '0320565500',
+        '70040041_0400500500' => '0400500500',
+        '60080000_0500500500' => '0901581400',
+        '60040071_0500500500' => '0512700600'
+      }.freeze
+
+      def converted_details
+        if @bank_code == '50040033'
+          msg = "Bank code #{@bank_code} is not used for payment transactions"
+          raise UnsupportedAccountDetails, msg
+        end
+
+        updated_account_number =
+          if PSEUDO_ACCOUNT_NUMBER_MAPPING.key?(combined_bank_details)
+            converted_pseudo_account_number
+          else
+            padded_account_number_for_validity
+          end
+
+        if EXCEPTION_BANK_CODES.include?(@bank_code) &&
+           (998000000..999499999).cover?(updated_account_number.to_i)
```

Comment:
Would `updated_account_number.to_i.between?(998000000, 999499999)` be a bit clearer? That's what we settled on for modulus checking Germany